### DISCUSSION TITLE
Is it possible to cancel a refund?

### DISCUSSION THREAD

**Discussion by sd-hi** [dojo-engineering/feedback#11] [Q&A] [answered]
Hi Dojo team,

For the Dojo Payments API, is there a mechanism to cancel/reverse a Refund that has been created against a captured payment intent?

I am working on integrating Dojo with an ERP system that has a refund workflow that is symmetrical to the sales workflow.

For example we 'create then capture' the payment intent and 'reverse' the payment intent if something goes wrong between ~~creation~~ authorization and capture (e.g. user abandons the checkout process because the customer decides to pay with another method).

Similarly, it would be ideal to create a 'refund' and then 'cancel' the refund if something went wrong (e.g. user abandons refund checkout process, because the customer decides to request a refund via different means).

Thanks in advance!

_Edit: Note for future readers, I am using captureMode:Manual_

**Comment by GabrielBoceanuDojo** [comment, marked as answer]
Hi,

Before I can correctly answer your question, I just want to clarify some things first:

1. In paragraph 3 above, what do you mean by **"reverse"**? Do you mean cancelling a payment which was not captured?
Because for our API, reversal has a very specific purpose, which is to "reverse" an already captured payment: https://docs.dojo.tech/payments/api#tag/Reversal (reversal can only be called on the day when the payment was captured, after that you will have to use the refund flow.)

2. In paragraph 4. Once the refund is made via API and it was successful, there is no real way to cancel it.
Although I am not familiar with the details of how your system is implemented, but for the specific refund flow, my advice would be to make the refund request only at very last stage of your "refund checkout" process.

Thanks,
Gabriel

**Reply by sd-hi** [reply]
Hi Gabriel,

Thanks for the quick response.

1. I am using the `captureMode:Manual` so the payment intent has a status of `Authorized` when performing the Reversal (unless they haven't filled in the payment page yet, in which case I do a cancel AKA HTTP DELETE on the payment intent). I've updated my original post to avoid confusing future readers.

2. Thanks for confirming, I will see if I can adjust our workflow to push the refund to the very last step. By the way, does the API offer a way to verify the remaining amount available for refund on a given payment intent? While we do have refund checks of our own, I think it would help our workflow to verify that our refund attempt is likely to succeed on the other end.

**Reply by RomanKondratyev** [reply]
Hi sd-hi,

Regarding your second question, you can check the `refundedAmount` proprerty of the PI in question.

Thanks,
Roman

**Comment by RomanKondratyev** [comment]
Regarding cancellation of a refund, we currently do not support this feature. However, we are going to discuss this and include it later.
